core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -138,6 +138,14 @@ class SparkHadoopUtil extends Logging {
   }
 
   /**
+   * Update credentials manually. This will trigger AM to renew the credentials immediately,
+   * also executors and driver (in client mode) will update the credentials accordingly.
+   *
+   * Note this will only be worked under YARN cluster manager.
+   */
+  def updateCredentials(sc: SparkContext): Unit = { }
```

Comment:
You might want to return a Future for this, which can indicate when the credentials have been updated : as of now, developers invoking this api have no way to know when (if at all) the credentials have been propagated through the driver/executors.